tests/snuba/api/endpoints/test_organization_tagkey_values.py

```diff
@@ -555,3 +559,95 @@ def test_invalid_keys(self):
     def test_boolean_fields(self):
         self.run_test("error.handled", expected=[("true", None), ("false", None)])
         self.run_test("error.unhandled", expected=[("true", None), ("false", None)])
+
+
+@region_silo_test
+class ReplayOrganizationTagKeyValuesTest(OrganizationTagKeyTestCase, ReplaysSnubaTestCase):
+    def setUp(self):
+        super().setUp()
+        replay1_id = uuid.uuid4().hex
+        replay2_id = uuid.uuid4().hex
+        replay3_id = uuid.uuid4().hex
+        seq1_timestamp = datetime.datetime.now() - datetime.timedelta(seconds=22)
+        self.store_replays(
+            mock_replay(
+                seq1_timestamp,
+                self.project.id,
+                replay1_id,
+                urls=[
+                    "http://localhost:3000/",
+                    "http://localhost:3000/login",
+                ],  # duplicate urls are okay,
+                tags={"fruit": "orange"},
+                segment_id=0,
+            ),
+        )
+        self.store_replays(
+            mock_replay(
+                seq1_timestamp,
+                self.project.id,
+                replay1_id,
+                urls=[
+                    "http://localhost:3000/",
+                    "http://localhost:3000/login",
+                ],  # duplicate urls are okay,
+                tags={"fruit": "orange"},
+                segment_id=1,
+            ),
+        )
+        self.store_replays(
+            mock_replay(
+                seq1_timestamp,
+                self.project.id,
+                replay2_id,
+                urls=[
+                    "http://localhost:3000/",
+                    "http://localhost:3000/login",
+                ],  # duplicate urls are okay,
+                tags={"fruit": "orange"},
+            )
+        )
+        self.store_replays(
+            mock_replay(
+                seq1_timestamp,
+                self.project.id,
+                replay3_id,
+                urls=[
+                    "http://localhost:3000/",
+                    "http://localhost:3000/login",
+                ],
+                tags={"fruit": "apple", "drink": "water"},
+            )
+        )
+
+    def run_test(self, key, expected, **kwargs):
+        # all tests here require that we search in replays so make that the default here
+        qs_params = kwargs.get("qs_params", {})
+        qs_params["includeReplays"] = "1"
+        kwargs["qs_params"] = qs_params
+        super().run_test(key, expected, **kwargs)
+
+    def test_simple(self):
+        self.run_test("fruit", expected=[("orange", 2), ("apple", 1)])
+        self.run_test("replay_type", expected=[("session", 3)])
+        self.run_test("environment", expected=[("production", 3)])
+        self.run_test("dist", expected=[("abc123", 3)])
+        self.run_test(
+            "url", expected=[("http://localhost:3000/login", 3), ("http://localhost:3000/", 3)]
+        )
```

Comment:
Since its not intuitive how arrayJoin is affecting the query could we add more diverse mocks for this case?  Might be good for catching regressions.